virtualhome/grails-app/views/templates/_passwordinput.gsp

```diff
@@ -77,16 +77,16 @@
     }
   }, "");
 
-  jQuery.validator.addMethod("noSpace", function(value, element) { 
-    return value.indexOf(" ") < 0 && value != ""; 
-  }, "Spaces are not permitted.");
+  jQuery.validator.addMethod("matchesPattern", function(value, element) {
+    return value.match(/^[_a-zA-Z][-_.a-zA-Z0-9]*$/) && value != "";
+  }, "Not a permitted username.");
 
   $("#accountform").validate({
     rules: {
       login: {
         required: true,
         minlength: 3,
```

Comment:
```suggestion
        minlength: 5,
```

Reply:
Maybe also add `maxLength: 100` since that appears to be a restriction in the model.